### PR TITLE
Implement partial SVG generation

### PR DESCRIPTION
**Pull request by gferon**
The idea is to do wireframe for ways + parks and waterways.

We should use this PR to discuss of API changes we want to introduce, to reduce the amount of boilerplate code necessary to use `osmflat`.

On top of my head:
* Iterator to loop / match through tags
* Maybe not use assert in `at()` and make it rusty by returning an Option?

### CODE REVIEW

File: examples/render-svg.rs [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+            next_element,
+        }
+    }
+}
+
+/// Iterator over osmflat nodes.
+///
+/// A polyline contains ranges of nodes. This iterator iterates over the ranges
+/// and resolves nodes as GeoCoord's.
+struct PolylineIter {
```
Comment (VeaaC) [inline examples/render-svg.rs]: Could you replace this iterator with a map of flattened ranges? (e.g. just a function that creates you the transform). Could be much more concise.

File: examples/render-svg.rs [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+        if (key == "leisure" && val == "park")
+            || (key == "landuse" && (val == "recreation_ground" || val == "forest"))
+        {
+            is_park = true;
+        }
+        if key == "water" && val == "lake" {
+            is_lake = true;
+        }
+
+        if is_multipolygon {
```
Comment (VeaaC) [inline examples/render-svg.rs:260]: Finish iterating all tags before deciding?

File: examples/render-svg.rs (hunk 1 above)
Comment (boxdot) [reply examples/render-svg.rs]: Good idea. Done.

File: examples/render-svg.rs (hunk 2 above)
Comment (boxdot) [reply examples/render-svg.rs:260]: I don't see why I can't do an early abort.

File: examples/render-svg.rs [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+            _ => None,
+        })
+        .collect();
+    Polyline { inner }
+}
+
+/// Classifies all features from osmflat we want to render.
+fn classify<'a>(archive: &'a osmflat::Osm) -> impl Iterator<Item = Feature> + 'a {
+    let inner_archive = archive.clone();
+    let ways = (0..archive.ways().len() as u64 - 2)
```
Comment (VeaaC) [inline examples/render-svg.rs:173]: Why -2?
The last is a sentinel, but shouldn't it be just -1?

File: examples/render-svg.rs [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+        .filter_map(move |idx| classify_way(&inner_archive, idx).map(|cat| Feature { idx, cat }));
+    let rels = (0..archive.relations().len() as u64 - 2)
```
Comment (VeaaC) [inline examples/render-svg.rs:175]: Same here?

**Pull request merged**